src/Core/Components/List/ListComponentBase.razor.cs

```diff
@@ -550,67 +571,20 @@ protected virtual async Task RaiseChangedEventsAsync()
                 builder.AddContent(0, ChildContent);
             });
         }
-
-        EventCallback<string> OnSelectCallback(TOption? item)
-        {
-            return EventCallback.Factory.Create<string>(this, (e) => OnSelectedItemChangedHandlerAsync(item));
-        }
-    }
-
-    /// <summary />
-    protected virtual async Task OnChangedHandlerAsync(ChangeEventArgs e)
-    {
-        if (e.Value is not null && Items is not null && !Multiple)
-        {
-            TOption? item = Items.FirstOrDefault(i => GetOptionValue(i) == e.Value.ToString());
-
-            await OnSelectedItemChangedHandlerAsync(item);
-        }
-        if (e.Value is not null && Multiple)
-        {
-            InternalValue = e.Value.ToString();
-        }
     }
 
     protected virtual async Task OnKeydownHandlerAsync(KeyboardEventArgs e)
     {
-        if (e is null || Multiple || Items is null)
+        if (e is null || Multiple)
         {
             return;
         }
 
+        string id = await Module!.InvokeAsync<string>("getAriaActiveDescendant", Id);
 
-        TOption? item = default;
-
-        // Get the previous and next options based on selected option
-        // Element at 0:  Previous, 1: Current, 2: Next
-        List<TOption?>? sandwichedItems = FindPrevAndNextOptions().ToList();
-
-        switch (e.Key)
-        {
-            case "Home":
-                item = Items.FirstOrDefault();
-                break;
-            case "ArrowDown":
-            case "NumpadArrowDown":
-                item = sandwichedItems[2];
-                break;
-            case "ArrowUp":
-            case "NumpadArrowUp":
-                item = sandwichedItems[0];
-                break;
-            case "End":
-                item = Items.LastOrDefault();
-                break;
-            case "Enter":
-            case "Escape":
-            default:
-                break;
-        }
-
-        await OnSelectedItemChangedHandlerAsync(item);
-
+        FluentOption<TOption> item = _internalListContext.Options.First(i => i.Id == id);
 
+        await item.OnClickHandlerAsync();
```

Comment:
Where is the code for using the arrow keys (Up, Down, Home, Escape), included in the previous code?

Reply:
It is not needed anymore as it is already handled by the web component